Hi3Helper.Core/Classes/Shared/Region/LauncherConfig.cs

```diff
@@ -266,6 +267,12 @@ public static bool IsShowRegionChangeWarning
             set => SetAndSaveConfigValue("ShowRegionChangeWarning", value);
         }
 
+        public static bool IsInstantRegionChange
+        {
+            get => GetAppConfigValue("UseInstantRegionChange").ToBool();
+            set => SetAndSaveConfigValue("UseInstantRegionChange", value);
+        }
+
```

Comment:
check `IsInstantRegionNeedRestart` and current mode before we return the config value? need to save the config values at startup.

Reply:
I'm not sure if that's necessary

For the comment body about can't change region when feature is disabled before restart, i fixed it with this 

```csharp
        private static bool? _cachedIsInstantRegionChange = null;
        public static bool IsInstantRegionChange
        {
            get
            {
                _cachedIsInstantRegionChange ??= GetAppConfigValue("UseInstantRegionChange").ToBool();
                return (bool)_cachedIsInstantRegionChange;
            }
            set => SetAndSaveConfigValue("UseInstantRegionChange", value);
        }
```

Reply:
Fixed in https://github.com/CollapseLauncher/Collapse/pull/449/commits/dd2901c974e260044f78024631e13d69988b9e12